src/SonarScanner.MSBuild.PreProcessor/JreCaching/JreCache.cs

```diff
@@ -141,7 +141,27 @@ private static void EnsureClosed(Stream fileStream)
 
     private async Task<JreCacheResult> UnpackJre(string downloadTarget, JreDescriptor jreDescriptor, string cacheRootLocation)
     {
-        return new JreCacheFailure("NotImplemented. The JRE is downloaded, but we still need to check, unpack, and set permissions.");
+        if (ValidateChecksum(downloadTarget, jreDescriptor.Sha256))
```

Comment:
This check should actually already be done on the temporary file before the rename (we also need to make sure that the check is done if the download is skipped).